server/router/src/main/java/org/infinispan/server/router/router/impl/hotrod/HotRodRouter.java

```diff
@@ -0,0 +1,101 @@
+package org.infinispan.server.router.router.impl.hotrod;
+
+import io.netty.bootstrap.ServerBootstrap;
+import io.netty.buffer.PooledByteBufAllocator;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelOption;
+import io.netty.channel.epoll.EpollServerSocketChannel;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.ServerSocketChannel;
+import io.netty.channel.socket.nio.NioServerSocketChannel;
+import io.netty.util.concurrent.DefaultThreadFactory;
+import org.infinispan.commons.logging.LogFactory;
+import org.infinispan.server.router.RoutingTable;
+import org.infinispan.server.router.configuration.HotRodRouterConfiguration;
+import org.infinispan.server.router.logging.RouterLogger;
+import org.infinispan.server.router.router.Router;
+import org.infinispan.server.router.router.impl.hotrod.handlers.SniHandlerInitializer;
+
+import java.lang.invoke.MethodHandles;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.util.Optional;
+
+public class HotRodRouter implements Router {
+
+    private static final RouterLogger logger = LogFactory.getLog(MethodHandles.lookup().lookupClass(), RouterLogger.class);
```

Comment:
`MethodHandles` are the simplest way to get current class within a static context (note that it's a static field). I often use this mechanism and embed it into my IntelliJ templates. This way I don't need to care to which class this logger belongs to.

However if this bothers you (I assure there is no performance penalty of this), I can replace it with `HotRodRouter.class` but I don't like this approach since renaming the class may cause compile errors (e.g. IDE is not clever enough to figure out that the logger initialization needs to be changed as well). 

So in my opinion this approach has more advantages than disadvantages...